test/functional/users_controller_test.rb

```diff
@@ -364,6 +365,39 @@ class UsersControllerTest < ActionController::TestCase
     assert_response :unprocessable_entity
   end
 
+  context "an active session is present" do
+    test "#login redirects to root(dashboard)" do
+      get :login, set_session_user
+
+      get :login
+      assert_redirected_to root_path
+    end
+
+    test "for user :one -> #login redirects to root(dashboard)" do
+      get :login, set_session_user.merge(user: users(:one).id)
+
+      get :login
+      assert_redirected_to root_path
+    end
+  end
+
+  context "no active session present" do
+    test "#login renders login page" do
```

Comment:
Sorry, I missed this point. One assertion per test isn't something we try to implement, it's OK if it's asserting multiple things as the test environment/request is identical.

Consider adding messages or comments to the assertions if it's unclear why they're testing what they do, but generally it's clearer to have one test block testing different things about the same request rather than spreading it out.

(So if you'd like to add the `assert_template :login` back to the existing test, that'd be fine.)